mergify_engine/context.py

```diff
@@ -745,31 +745,58 @@ async def _get_consolidated_data(self, name):
             return [r["user"]["login"] for r in comments if r["state"] == "COMMENTED"]
 
         # NOTE(jd) The Check API set conclusion to None for pending.
-        # NOTE(sileht): "pending" statuses are not really trackable, we
-        # voluntary drop this event because CIs just sent they status every
-        # minutes until the CI pass (at least Travis and Circle CI does
-        # that). This was causing a big load on Mergify for nothing useful
-        # tracked, and on big projects it can reach the rate limit very
-        # quickly.
-        # NOTE(sileht): Not handled for now: cancelled, timed_out, or action_required
         elif name in ("status-success", "check-success"):
             return [
                 ctxt
                 for ctxt, state in (await self.checks).items()
                 if state == "success"
             ]
         elif name in ("status-failure", "check-failure"):
+            # hopefully "cancelled" is actually a failure state to github.
+            # I think it is, however it could be the same thing as the
+            # "skipped" status.
             return [
                 ctxt
                 for ctxt, state in (await self.checks).items()
                 if state == "failure"
+                or state == "cancelled"
+                or state == "timed_out"
+                or state == "action_required"
             ]
         elif name in ("status-neutral", "check-neutral"):
             return [
                 ctxt
                 for ctxt, state in (await self.checks).items()
                 if state == "neutral"
             ]
+        elif name in ("status-skipped", "check-skipped"):
+            # hopefully this handles the gray "skipped" state that github actions
+            # workflows can send when a job that depends on a job and the job it
+            # depends on fails, making it get skipped automatically then.
+            return [
+                ctext
+                for ctext, state in (await self.checks).items()
+                if state == "skipped"
+            ]
+        elif name in ("status-pending", "check-pending"):
+            return [
+                ctext
+                for ctext, state in (await self.checks).items()
+                if state != "success"
+                and state != "failure"
+                and state != "cancelled"
+                and state != "timed_out"
+                and state != "action_required"
+                and state != "neutral"
+                and state != "skipped"
+                and state != "stale"
```

Comment:
GitHub may continue to add new states and we don't want to put them in pending list by default. I think just checking `state in [None, "pending"]` should be enough.

Reply:
Alright changed it locally, also changed the other ones to be similar on the multiple status checks to be more easier to read.